File: python_backend/game_logic/fenParser/GameStatus/get_status_str.py
import re
import os


def get_status_str(fen):
    """method to strip the piece positions part of fen string, and return the remaining data"""
    status_str = re.sub(r'([a-zA-Z1-8]+/*){8}', '', fen)
    status_str.strip()
    return status_str


if __name__ == "__main__":
    saved_games = os.listdir('../../saved games')
    for game in saved_games:
        f = open('../../saved games/{}/{}.fen'.format(game, game), 'r')
        fen = f.readline()
        print(get_status_str(fen))

File: app/src/game_logic/fenParser/GameStatus/getStatusStr.js


export function getStatusStr(fen) {
    /*method to strip the piece positions part of fen string, &&  return the remaining data**/
    let statusStr = fen.replace(/([a-zA-Z1-8]+\/){7}([a-zA-Z1-8]+)/gi, '')
    statusStr = statusStr.trim()
    return statusStr
}

// '8/8/3k4/rnbq1bnr/8/RNBQ1BNR/8/1RBRK3 w - - 0 1'

// module.exports = getStatusStr;

/** for node.js
if (require.main === module) {


}
*/
